impl/src/main/java/org/apache/myfaces/renderkit/RendererUtils.java

```diff
@@ -693,17 +693,25 @@ else if (allowNonArrayOrCollectionValue)
     public static Object getConvertedUISelectOneValue(
             FacesContext facesContext, UISelectOne output, Object submittedValue)
     {
-        if (submittedValue != null && !(submittedValue instanceof String))
+
+        if(submittedValue == null)
+        {
+            if (log.isLoggable(Level.FINE))
+            {
+                log.fine("No conversion necessary for null uiselectone value: client id " + output.getClientId());
+            }
+            return null;
```

Comment:
Mojarra returns null and so I did the same here ( since no value is submitted)

I'm just not sure if this is correct, however.

Reply:
Might be worth a challenge? Unless we switch to enums which would signify that no value is inputted, hence no conversion necessary (rather than rely on this empty string / null nonsense).

Reply:
Might be a good question in a challenge for BalusC and Arjan to answer?